video/clip.go

```diff
@@ -89,67 +97,109 @@ func ConvertUnixMillisToSeconds(requestID string, firstSegment *m3u8.MediaSegmen
-			return nil, fmt.Errorf("error clipping: failed to get an ending index")
+			return nil, []ClipSegmentInfo{}, fmt.Errorf("error clipping: failed to get an ending index")
 		}
 	}
 
+	firstSegmentToClip := clipStartSegmentInfo.SequenceID
+	lastSegmentToClip := clipEndSegmentInfo.SequenceID
 	// Generate a slice of all segments that overlap with startTime to endTime
-	relevantSegments := manifest.Segments[startSegIdx : endSegIdx+1]
+	relevantSegments := manifest.Segments[firstSegmentToClip : lastSegmentToClip+1]
 	totalRelSegs := len(relevantSegments)
 	if totalRelSegs == 0 {
-		return nil, fmt.Errorf("error clipping: no relevant segments found in the specified time range")
+		return nil, []ClipSegmentInfo{}, fmt.Errorf("error clipping: no relevant segments found in the specified time range")
 	}
 
-	log.Log(requestID, "Clipping segments", "from", startSegIdx, "to", endSegIdx)
+	log.Log(requestID, "Clipping segments", "from", firstSegmentToClip, "to", lastSegmentToClip)
 
-	return relevantSegments, nil
+	// If the clip start/end times fall within the same segment, then
+	// save only the single segment's info
+	var cs []ClipSegmentInfo
+	if firstSegmentToClip == lastSegmentToClip {
+		cs = []ClipSegmentInfo{clipStartSegmentInfo}
+	} else {
+		cs = []ClipSegmentInfo{clipStartSegmentInfo, clipEndSegmentInfo}
+	}
+	return relevantSegments, cs, nil
 }
 
-// Clips a segment by re-encoding so that I-frames are placed at the right intervals
-// Currently using these settings for the encode step:
+// Clips a segment from the specified start/end times. In some cases, the segment
+// will be re-encoded so that I-frames are placed at the right intervals.
+// This allows for frame-accurate clipping.
+// Currently using a combination of these settings for the encode step:
 //
 //	"c:v": "libx264": Specifies H.264 video codec.
-//	"r": "24": Sets video frame rate to 24 FPS.
 //	"g": "48": Inserts keyframe every 48 frames (GOP size).
 //	"keyint_min": "48": Minimum keyframe interval.
 //	"sc_threshold": 50: Detects scene changes with threshold 50.
 //	"bf": "0": Disables B-frames for bidirectional prediction.
-//	"c:a": "copy": Copies audio stream without re-encoding.
+//	"c:a": "aac": re-encode audio and clip.
 func ClipSegment(tsInputFile, tsOutputFile string, startTime, endTime float64) error {
-	start := formatTime(startTime)
-	end := formatTime(endTime)
-	err := ffmpeg.Input(tsInputFile).
-		Output(tsOutputFile, ffmpeg.KwArgs{"ss": start, "to": end,
+	var start, end string
+	var args ffmpeg.KwArgs
+	if endTime < 0 {
+		// Clip from specified start time to the end of
+		// the segment (when clipping starting segment)
+		start = formatTime(startTime)
+		args = ffmpeg.KwArgs{"bf": "0",
+			"c:a":          "aac",
 			"c:v":          "libx264",
-			"r":            "24",
 			"g":            "48",
 			"keyint_min":   "48",
 			"sc_threshold": 50,
-			"bf":           "0",
-			"c:a":          "copy"}).
+			"ss":           start}
+	} else if startTime < 0 {
```

Comment:
Is it definitely `< 0` and not `<=0` just double checking, is it set to -1 or something somewhere?

Reply:
Yup, see this: 
https://github.com/livepeer/catalyst-api/pull/890/files#diff-47d60fc2d61cc45a41ecad1e68397953b10fe4d7d8dcd3e404b5568263fbd8c2R298

I'm using it as an override to the ClipSegment function to clip with different ffmpeg calls. I don't *love* how I did this so will refactor and clean it up in a follow-up. I think having separate functions with different ffmpeg calls would be more readable here.